### DISCUSSION TITLE
Calculate individual class probabilities

### DISCUSSION THREAD

**Discussion by HareeshBahuleyan** [Raschka-research-group/coral-pytorch#27] [Q&A]
Thanks for sharing this useful library. The instructions on the documentation page are clear and easy to follow.

My question is regarding the output probabilities. As I understand, both coral and corn outputs cumulative probabilities. For instance, if we are dealing with a 5-class (classes 0 to 4) problem, then the 4 probabilities output are: 
```
[p(y > 0), p(y > 1), p(y > 2), p(y > 3)]
```

Is there a way to obtain individual class probabilities from this, such as p(y=1), p(y=2) and so on?

Would you say the following is valid to do?
```
P(0) = 1 - P(y>0) 
P(1) = P(y>0) - P(y>1)
P(2) = P(y>1) - P(y>2)
P(3) = P(y>2) - P(y>3)
P(4) = P(y>3)
```

**Comment by rasbt** [comment]
Yes, this should be valid.

**Reply by HareeshBahuleyan** [reply]
Thanks @rasbt for the reply.

For my use case when I applied this, there seemed to be a mismatch between the label calculated via `argmax` over the individual probabilities obtained using the above formulae vs. the predicted label obtained via `proba_to_label`.

Here is an example:

**Method 1: Using argmax over individual probabilities**

```
p(y>0)	0.72
p(y>1)	0.55
p(y>2)	0.32
p(y>3)	0.25
```

```
p(y=0)	0.28
p(y=1)	0.17
p(y=2)	0.23
p(y=3)	0.07
p(y=4)	0.25
---
p(total) 1.0
```

_index of predicted class via `argmax` ->	0_

**Method 2: Using `proba_to_label`**

```
probas = [0.72, 0.55, 0.32, 0.25]
predict_levels = probas > 0.5 # [1, 1, 0, 0]
predicted_labels = torch.sum(predict_levels, dim=1) # 2
```

_index of predicted class via `proba_to_label ` ->	2_

Is there some that I might be missing here? Or do you know why this discrepancy occurs?

**Reply by rasbt** [reply]
That's a good point. Both are slightly different ways for obtaining the final label and they don't have to necessarily produce the same label all the time. E.g., it could be that you have something like [0.6, 0.5, 0.4, 0.1]. In this case the approach we use in the paper yields label 3 because of sum([1, 1, 0, 0])+1) and via argmax on 

P(0) = 1 - P(y>0) = 0.4   
P(1) = P(y>0) - P(y>1) = 0.1 
P(2) = P(y>1) - P(y>2)  = 0.1
P(3) = P(y>2) - P(y>3)  = 0.3

you'd get label 0.

**Reply by HareeshBahuleyan** [reply]
Understood, thanks for the clarification.
